### PR TITLE
Add support to PHP 8.3

### PR DESCRIPTION
**Pull request by roberto-butti**
Hi everyone,
I tried to update the dependencies for PHP 8.3.
I also update the GitHub actions workflow for running tests with php8.3
One note about phpunit10.2: because phpunit10.2 introduces some breaking changes, I added 
```
 "phpunit/phpunit": ">= 10.1.2 <10.2"
```
in the composer. json.
But this is a new issue (not strictly related to PHP8.3), which is why another PR should address the support for phpunit10.2.

Please feel free to give me any feedback you might have.

Roberto

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -43,7 +43,7 @@
         "phpunit/php-code-coverage": "^10.1.1",
         "phpunit/php-file-iterator": "^4.0.1",
         "phpunit/php-timer": "^6.0",
-        "phpunit/phpunit": "^10.1.2",
+        "phpunit/phpunit": ">= 10.1.2 <10.2",
```
Comment (Slamdunk) [inline composer.json]: I've added expicit support for PHPUnit 10.2 in https://github.com/paratestphp/paratest/pull/767: can you rebase a remove this line please?

**Review by Slamdunk — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (roberto-butti) [reply composer.json]: Hi @Slamdunk  , rebased and applied the changes according to your request.
Let me know if it is ok from your side.

Comment (roberto-butti) [reply composer.json]: Hi @Slamdunk , rebased and applied the changes. Tests from php 8.1 to 8.3 are green ✅

Comment (Slamdunk) [reply composer.json]: Mmm, something went wrong.

I see you didn't create a separate branch for this changes, so I don't know what happened when you say "_rebased and applied the changes_".

If the rebase had went as expected, we wouldn't see the changes in `src/WrapperRunner/ApplicationForWrapperWorker.php` perhaps.

Can you try moving the edits in a separate branch, rebase it onto the `upstream/7.x` e push it here please?

Comment (roberto-butti) [reply composer.json]: Rebasing the main branch, i missed the "Handle unexpected output (#769)".
Now as you can see, the small differences (with php8.3) between my branch and the upstream are:

https://github.com/paratestphp/paratest/pull/766/files

**Review by Slamdunk — APPROVED** [review]

**Comment by Slamdunk** [comment]
Thank you

**Pull request merged**